src/index.js

```diff
@@ -0,0 +1,69 @@
+///////////////////////////////////////////////////////////////////////////////////
+// The MIT License (MIT)
+//
+// Copyright (c) 2017 Tarek Sherif
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy of
+// this software and associated documentation files (the "Software"), to deal in
+// the Software without restriction, including without limitation the rights to
+// use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+// the Software, and to permit persons to whom the Software is furnished to do so,
+// subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+// FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+// COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+// IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+// CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+///////////////////////////////////////////////////////////////////////////////////
+
+"use strict";
+
+var App    = require('./app');
+var PicoGL = require('./picogl');
+
+
+/**
+    Global PicoGL module. For convenience, all WebGL enums are stored
+    as properties of PicoGL (e.g. PicoGL.FLOAT, PicoGL.ONE_MINUS_SRC_ALPHA).
+
+    @namespace PicoGL
+    @prop {string} version Current PicoGL version.
+    @prop {object} TEXTURE_INTERNAL_FORMAT Map of framebuffer texture formats to internal formats.
+    @prop {object} TYPE_SIZE Map of data types to sizes in bytes.
+    @prop {object} WEBGL_INFO WebGL context information.
+*/
+global.PicoGL = PicoGL;
+
+PicoGL.version = "<%= VERSION %>";
+PicoGL.App = App;
+PicoGL.Program = require('./program');
+PicoGL.Texture = require('./texture');
+PicoGL.Cubemap = require('./cubemap');
+PicoGL.DrawCall          = require('./drawcall');
+PicoGL.Framebuffer       = require('./framebuffer');
+PicoGL.Shader            = require('./shader');
+PicoGL.Timer             = require('./timer');
+PicoGL.TransformFeedback = require('./transformfeedback');
+PicoGL.UniformBuffer     = require('./uniformbuffer');
+PicoGL.VertexArray       = require('./vertexarray');
+PicoGL.VertexBuffer      = require('./vertexbuffer');
+
+
+/**
+    Create a PicoGL app. The app is the primary entry point to PicoGL. It stores
+    the canvas, the WebGL context and all WebGL state.
+
+    @function PicoGL.createApp
+    @param {DOMElement} canvas The canvas on which to create the WebGL context.
+    @param {Object} [contextAttributes] Context attributes to pass when calling getContext().
+*/
+PicoGL.createApp = function(canvas, contextAttributes) {
```

Comment:
I moved everything out of `picogl.js` _except_ for all of the gl constants. That way all the modules in picogl that need access to the constants can just `require('./picogl')` and access them rather than relying on any global variables.